src/Mafia/Cabal/Package.hs

```diff
@@ -67,6 +68,27 @@ getCabalFile dir = do
     _ : _ ->
       left $ CabalMultipleFilesFound dir xs
 
+-- | Given a file or directory path, try to find the .cabal file by searching
+--   its parent directories.
+findCabalRoot :: MonadIO m => Path -> m (Maybe Directory)
+findCabalRoot path0 = do
+  path <- canonicalizePath path0
+  entries <- tryGetDirectoryContents path
+  if any (\x -> extension ".cabal" x && x /= ".cabal" && not (T.isSuffixOf "-test.cabal" x)) entries then
```

Comment:
Could we get src and test from the cabal file? or would that slow it down too much

Hit a similar problem with hinty yesterday, burning in our search path conventions is v tempting